mix.exs

```diff
@@ -44,7 +44,8 @@ defmodule K8s.MixProject do
       {:ex_doc, "~> 0.20", only: :dev},
       {:excoveralls, "~> 0.10", only: :test},
       {:mix_test_watch, "~> 0.8", only: :dev, runtime: false},
-      {:stream_data, "~> 0.4", only: :test}
+      {:stream_data, "~> 0.4", only: :test},
+      {:websockex, "~> 0.4.0"}
```

Comment:
As far as I understand optional deps (`optional: true`) it will still work as usual but if someone else includes k8s they have the option to either add the optional or not. It is not forced on them. At first glance, this seems like it might be confusing to the person using the library as it wont "work out of the box". Worth testing though. What is your thought process around this?

Reply:
Even more confusing (if i have read the commit correctly), elixir versions < 1.8 do not start optional deps by default. They must be added to extra app (like we have here). Versions 1.8 >= start them by default.